.lefthook.yaml

```diff
@@ -0,0 +1,4 @@
+pre-commit:
+  commands:
+    secret-scan:
+      run: gitleaks protect --verbose --redact --staged
```

Comment:
If gitleaks isn't installed, the command should fail with a clear error message explaining to user what to do (ideally so we can just copy-paste). Maybe even have a Makefile target `make check-gitleaks` or something, that would both install the tool if it isn't installed, and run it too so it can be executed outside of the pre-commit hook?

Reply:
Implemented this. Doesn't auto install for for the user since the build process is ever so slightly not simple since it sets the version through `go build -ldflags ...`.

It will print out the installation instructions in the README and the compiled executables in the releases.